tests/framework/pkg/session/session.go

```diff
@@ -10,6 +10,7 @@ type Session struct {
 	cleanupQueue []CleanupFunc
 	open         bool
 	testingT     *testing.T
+	subSession   *Session
```

Comment:
I don't see why sessions would need to track the sub session? When a subsession is created it registers its cleanup with the parent.

Reply:
Yes, but at the test case scope our clients have no reference to the subsession so if we want to delete resources at the test case scope nothing will happen until we clear up our resources at the test suite scope. Does that make sense?

Reply:
when the test creates the sub session it creates a variable to hold it.  If the test session want to create a sub session it is responsible for calling the cleanup when appropriate.  If that test session fails to cleanup it will be called when the parent session in cleaned up.